src/lib/transaction.rs

```diff
@@ -39,6 +63,177 @@ pub struct TransactionPayload {
     pub(crate) call_data: Vec<u8>,
 }
 
+impl TransactionPayload {
+    pub fn new(
+        anchor: BlsScalar,
+        nullifiers: Vec<BlsScalar>,
+        crossover: Option<Crossover>,
+        notes: Vec<Note>,
+        fee: Fee,
+        spending_proof: Proof,
+        call_data: Vec<u8>,
+    ) -> Self {
+        Self {
+            anchor,
+            nullifiers,
+            crossover,
+            notes,
+            fee,
+            spending_proof,
+            call_data,
+        }
+    }
+
+    pub fn into_bytes(&self) -> Vec<u8> {
```

Comment:
Same as mentioned above.

Reply:
Fixed, passing `&self` and naming `to_bytes`